File: server/appointment-service/src/main/java/com/lukemacdonald/appointmentservice/model/HealthStatus.java
package com.lukemacdonald.appointmentservice.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class HealthStatus{

    @Id
    private int appointmentId;

    @Column
    private Boolean feverOrChills;

    @Column
    private Boolean coughing;

    @Column
    private Boolean headaches;

    @Column
    private Boolean vomiting;

    @Column
    private Boolean fainting;

    @Column
    private String description;

}

File: client/lib/models/health_status.dart
class HealthStatus {

  late int id;
  bool fever = false;
  bool cough= false;
  bool headache = false;
  bool vomiting= false;
  bool faint= false;
  String description = "";

  setDetails(var details){
    id = details['appointmentId'];
    fever = details['feverOrChills'];
    cough= details['coughing'];
    headache = details ['headaches'];
    faint = details ['fainting'];
    vomiting = details['vomiting'];
    description = details['description'];
  }

}